Lares/Assets/Mechanics/Inventory/Runtime/InventoryItem.cs

```diff
@@ -0,0 +1,36 @@
+using System;
+using UnityEngine;
+
+namespace Lares.Inventory
+{
+    public class ObjectID : PropertyAttribute { }
+
+    [Serializable]
+    public struct InventoryData
+    {
+        [ObjectID] public string ItemId;
+        public string ItemName;
+        public string ItemDescription;
+        public Sprite ItemImage;
+    }
+
+    [Serializable]
+    public class InventoryItem : IItem
+    {
+        public InventoryData ItemData;
+        
+        public virtual void OnUse(GameObject playerRef) { return; }
+
+        public static bool operator ==(InventoryItem a, InventoryItem b) { return a.ItemData.ItemId == b.ItemData.ItemId; }
+        public static bool operator !=(InventoryItem a, InventoryItem b) { return a.ItemData.ItemId != b.ItemData.ItemId; }
+
+        public override bool Equals(System.Object obj)
+        {
+            if (obj == null) return false;
+            if (!(obj is InventoryItem)) return false;
+            return ItemData.ItemId == (obj as InventoryItem).ItemData.ItemId;
+        }
```

Comment:
A direct cast here will always succeed - also `object` (code style) is used far more than `System.Object` even if `object` is an alias for it.
```suggestion
        public override bool Equals(object obj)
        {
            if (obj is null or not InventoryItem) return false;
            return ItemData.ItemId == ((InventoryItem) obj).ItemData.ItemId;
        }
```